lms/djangoapps/course_api/tests/test_api.py

```diff
@@ -0,0 +1,119 @@
+"""
+Test for course API
+"""
+
+from django.http import Http404
+from django.test import RequestFactory
+from lettuce import world
+
+from opaque_keys import InvalidKeyError
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import ToyCourseFactory
+
+from lms.djangoapps.course_api.api import course_detail, list_courses
+
+
+class CourseApiTestMixin(object):
+
+    expected_course_data = {
+        'blocks_url': '/api/courses/v1/blocks/?course_id=edX%2Ftoy%2F2012_Fall',
+        'course_image': u'/c4x/edX/toy/asset/just_a_test.jpg',
+        'description': u'A course about toys.',
+        'end': None,
+        'enrollment_end': None,
+        'enrollment_start': None,
+        'id': u'edX/toy/2012_Fall',
+        'name': u'Toy Course',
+        'number': u'toy',
+        'org': u'edX',
+        'start': u'2015-07-17T12:00:00Z',
+        'start_display': u'July 17, 2015',
+        'start_type': u'timestamp',
+    }
+
+    @classmethod
+    def setUpClass(cls):
+        super(CourseApiTestMixin, cls).setUpClass()
+        cls.request_factory = RequestFactory()
+
+    @staticmethod
+    def create_course():
```

Comment:
To help maintain the time required by our unit tests, instead of creating a new course for each Test Method, create it only once for each Test Class.  Do this using [SharedModuleStoreTestCase](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/tests/django_utils.py#L216).